### PR TITLE
CSHARP-3161: Default OCSP to 'off' for drivers that hard-fail when an OCSP responder is unavailable

### PR DESCRIPTION
**Pull request by MikalaiMazurenka**
https://jira.mongodb.org/browse/CSHARP-3161
https://evergreen.mongodb.com/version/5f18f62c3e8e867f291ef891

### CODE REVIEW

File: tests/MongoDB.Driver.Legacy.Tests/MongoServerSettingsTests.cs [hunk 1]
```diff
@@ -113,6 +113,33 @@ public void TestClone()
...
+        public void TestCloneTlsDisableCertificateRevocationCheck()
+        {
+            var connectionString = "mongodb://somehost/?tlsDisableCertificateRevocationCheck=true";
+            var builder = new MongoUrlBuilder(connectionString);
+            var url = builder.ToMongoUrl();
+            var settings = MongoServerSettings.FromUrl(url);
+
+            var clone = settings.Clone();
+
+            Assert.Equal(settings, clone);
```
Comment (DmitryLukyanov) [inline tests/MongoDB.Driver.Legacy.Tests/MongoServerSettingsTests.cs]: use a new format for test asserting

File: tests/MongoDB.Driver.Tests/MongoClientSettingsTests.cs [hunk 2]
```diff
@@ -398,12 +424,14 @@ public void TestFreezeInvalid()
...
             // set everything to non default values to test that all settings are converted
+            // with the exception of tlsDisableCertificateRevocationCheck because setting that with tlsInsecure is
+            // not allowed in a connection string
             var connectionString =
                 "mongodb://user1:password1@somehost/?appname=app1;authSource=db;authMechanismProperties=CANONICALIZE_HOST_NAME:true;" +
                 "compressors=zlib,snappy;zlibCompressionLevel=9;connect=direct;connectTimeout=123;ipv6=true;heartbeatInterval=1m;heartbeatTimeout=2m;localThreshold=128;" +
                 "maxIdleTime=124;maxLifeTime=125;maxPoolSize=126;minPoolSize=127;readConcernLevel=majority;" +
                 "readPreference=secondary;readPreferenceTags=a:1,b:2;readPreferenceTags=c:3,d:4;retryReads=false;retryWrites=true;socketTimeout=129;" +
-                "serverSelectionTimeout=20s;tls=true;tlsInsecure=true;waitqueuesize=130;waitQueueTimeout=131;" +
+                "serverSelectionTimeout=20s;tls=true;sslVerifyCertificate=false;waitqueuesize=130;waitQueueTimeout=131;" +
```
Comment (DmitryLukyanov) [inline tests/MongoDB.Driver.Tests/MongoClientSettingsTests.cs:434]: Why it was replaced? is this obsolete option really required?

Comment (DmitryLukyanov) [reply tests/MongoDB.Driver.Tests/MongoClientSettingsTests.cs:434]: I think the only test you need is `tlsDisableCertificateRevocationCheck` which is the only scope of your ticket. 
I will rely on your and Robert's opinion here

File: tests/MongoDB.Driver.Legacy.Tests/MongoServerSettingsTests.cs (hunk 1 above)
Comment (rstam) [reply tests/MongoDB.Driver.Legacy.Tests/MongoServerSettingsTests.cs]: True in general. Sometimes in a file that uses the old style asserts it is acceptable to match the style in the file.

**Review by rstam — APPROVED** [review]
LGTM

Dmitry's comments in the test files could be looked at, but I'm willing to LGTM either way.

File: tests/MongoDB.Driver.Legacy.Tests/MongoServerSettingsTests.cs (hunk 1 above)
Comment (MikalaiMazurenka) [reply tests/MongoDB.Driver.Legacy.Tests/MongoServerSettingsTests.cs]: Done.

File: tests/MongoDB.Driver.Tests/MongoClientSettingsTests.cs (hunk 2 above)
Comment (MikalaiMazurenka) [reply tests/MongoDB.Driver.Tests/MongoClientSettingsTests.cs:434]: In this case we have 3 initial options controlling one setting. I would rather test it when I have an opportunity.

**Comment by MikalaiMazurenka** [comment]
Ready for review!

**Review by DmitryLukyanov — APPROVED** [review]
LGTM

**Comment by MikalaiMazurenka** [comment]
Sqashed, rebased, pushed to master.

**Pull request closed without merging**